client/src/app/(onboarding)/signup/page.tsx

```diff
@@ -1,3 +1,208 @@
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [hasSubmitted, setHasSubmitted] = useState(false); // Tracks form submission
+
+  // Error state for form validation
+  const [error, setError] = useState({
+    email: "",
+    password: "",
+    confirmPassword: ""
+  });
+
+  // Function to handle form validation
+  const validateForm = () => {
+    const formErrors = { email: "", password: "", confirmPassword: "" };
+    let isValid = true;
+
+    if (!email) {
+      formErrors.email = "Email is required";
+      isValid = false;
+    }
+
+    if (!password) {
+      formErrors.password = "Password is required";
+      isValid = false;
+    }
+
+    if (password !== confirmPassword) {
+      formErrors.confirmPassword = "Passwords do not match";
+      isValid = false;
+    } else if (!confirmPassword) {
+      formErrors.confirmPassword = "Confirm password is required";
+      isValid = false;
+    }
+
+    setError(formErrors);
+    return isValid;
+  };
+
+  const handleSubmit = () => {
+    setHasSubmitted(true); // Set the form as submitted
+    if (validateForm()) {
+      console.log("Email: ", email);
+      console.log("Password", password);
+    } else {
+      console.log("Validation failed");
+    }
+  };
+
+  return (
+    <div className="borderline main-container">
+      {/* Let's get started section */}
+      <Typography variant="h4" sx={{ marginBottom: "20px" }}>
+        Let’s Get Started.
+      </Typography>
+
+      {/* Sign in with Google Button */}
+      <Button
+        className="borderline"
+        variant="outlined"
+        startIcon={<img src="https://www.google.com/favicon.ico" alt="Google" />}
+        sx={{
+          border: "4px solid #E0E4F2",
+          backgroundColor: "white",
+          width: "780px",
+          borderRadius: "8px",
+        }}
+      >
+        Sign up with Google
+      </Button>
+
+      {/* Divider with "Or" */}
+      <div className="divider">
+        <Divider
+          sx={{
+            width: "100px",
+            height: "1px",
+            backgroundColor: "#ccc",
+            marginRight: 2,
+          }}
+        />
+        <Typography
+          variant="h4"
+          sx={{
+            fontWeight: "bold",
+            fontSize: "1.25rem",
+            textAlign: "center",
+          }}
+        >
+          Or
+        </Typography>
+        <Divider
+          sx={{
+            width: "100px",
+            height: "1px",
+            backgroundColor: "#ccc",
+            marginLeft: 2,
+          }}
+        />
+      </div>
+
+      {/* Signup form */}
+      <div className="signup-form">
+        {/* Email field */}
+        <div className="text-field">
+          <Typography
+            variant="subtitle1"
+            sx={{ fontWeight: "bold", marginBottom: "8px" }}
+          >
+            Email {hasSubmitted && <span style={{ color: "red" }}>*</span>}
```

Comment:
Is it possible to change the theme for this one so it renders the red asterisk? If that is not possible, make a new component based off the MUI TextField and the TextField props? We would need to this manually each time this TextField component is rendered and it will be hard to maintain in the future

Reply:
fixed